### ISSUE TITLE
Orchestrator doesn't receive cluster changes through discovery

### ISSUE THREAD

**Issue by jkrcma** [presslabs/mysql-operator#122]
Chart version: `0.1.12`
Operator image version: `v0.1.12`

From `operator` log:

```
I1004 07:40:16.312659       1 sync.go:36] Syncing cluster: silly
I1004 07:40:17.513492       1 orc_reconciliation.go:40] Orchestrator reconciliation for cluster 'silly' started...
E1004 07:40:17.530144       1 orc_reconciliation.go:48] Fail to get cluster from orchestrator: [orc]: status: 500 msg: at error, json unmarshal error: unexpected end of JSON input, details: <nil>. Now tries to register nodes.
I1004 07:40:17.701285       1 utils.go:172] Endpoints for service 'silly-mysql-master' were up-to-date.
E1004 07:40:17.707812       1 controller.go:225] mysqlclusterController controller: Re-queuing item "mysql-operator/silly" due to error processing: sync error: Operation cannot be fulfilled on mysqlclusters.mysql.presslabs.org "silly": the object has been modified; please apply your changes to the latest version and try again
```

From `helper` container in mysql pod:

```
I1004 07:53:37.332505      12 apphelper.go:54] Configured orchestrator user...
I1004 07:53:37.334482      12 apphelper.go:60] Configured replication user...
I1004 07:53:37.336214      12 apphelper.go:66] Configured metrics exporter user...
I1004 07:53:37.336254      12 util.go:70] Orc request on: http://mysql-operator-orchestrator.mysql-operator/api/master/silly.mysql-operator
I1004 07:53:37.350705      12 util.go:53] Unmarshal data is: 
E1004 07:53:37.350759      12 util.go:197] Failed to connect to orc for finding master, err: [orc]: status: 500 msg: at error, json unmarshal error: unexpected end of JSON input, details: <nil>. Fallback to determine master by its id.
I1004 07:53:37.350811      12 apphelper.go:72] Configured topology...
I1004 07:53:37.376783      12 apphelper.go:77] Flag setup as complete...
```

From `orchestrator` log:

```
[martini] Started GET /api/master/silly.mysql-operator for 10.244.0.84
2018-10-04 07:53:37 ERROR Unable to determine cluster name
[martini] Completed 500 Internal Server Error in 7.107462ms
```

After I add the cluster to Orchestrator manually the `operator` log output doesn't change much:

```
I1004 07:56:43.034502       1 orc_reconciliation.go:40] Orchestrator reconciliation for cluster 'silly' started...
W1004 07:56:44.192475       1 orc_reconciliation.go:158] [getRecoveriesToAck]: Cluster is not ready for ack.
I1004 07:56:45.377957       1 utils.go:172] Endpoints for service 'silly-mysql-master' were up-to-date.
E1004 07:56:45.384547       1 controller.go:225] mysqlclusterController controller: Re-queuing item "mysql-operator/silly" due to error processing: sync error: Operation cannot be fulfilled on mysqlclusters.mysql.presslabs.org "silly": the object has been modified; please apply your changes to the latest version and try again
```

The last line is repeating in the log all over again.

`MysqlCluster` manifest:
```
apiVersion: v1
kind: Secret
metadata:
  name: silly
  namespace: mysql-operator
type: Opaque
data:
  ROOT_PASSWORD: dGVzdHRlc3Q=

---

apiVersion: mysql.presslabs.org/v1alpha1
kind: MysqlCluster
metadata:
  name: silly
  namespace: mysql-operator
spec:
  replicas: 3
  secretName: silly
```

**Comment by AMecea** [comment]
Hi @jkrcma, The operator registers the cluster when the first node is ready, then the orchestrator is in charge of discovering new nodes. Please give me more details about your cluster, more logs of the operator and cluster state. We can chat further on the [gitter](https://gitter.im/PressLabs/mysql-operator) channel to debug this issue together.

In the next version of mysql-operator, which is in progress right now, the operator will register/unregister nodes into orchestrator, this will solve this issue.

**Comment by AMecea** [comment]
We figure it out what is the cause for this issue. The operator does not update the status for `MysqlCluster` so nodes will not be registered in orchestrator. This is because status is declared as subresource and in k8s >= 1.11 is not updated.

The next version of the operator, that is in progress, will work on 1.11. Fixed by #102
